Wrappers/Python/cil/framework/data_container.py

```diff
@@ -464,7 +464,7 @@ def add(self, other, *args, **kwargs):
     def subtract(self, other, *args, **kwargs):
         if hasattr(other, '__container_priority__') and \
            self.__class__.__container_priority__ < other.__class__.__container_priority__:
-            return other.subtract(self, *args, **kwargs)
+            return (-other).add(self, *args, **kwargs)
```

Comment:
I'm not sure about this approach. Would is be better to just raise an error? This will increase the memory use, and will be hidden from the user so they would struggle to improve their code to avoid it.

Reply:
I suppose I lean towards "technically more correct even if more memory-intensive", and "it's up to the user to do hacky things like replace `c - x` with `c.subtract(x)`".

Could add a note in the `subtract` docstring I suppose? "Note this might `do (-other).add(self)"

Reply:
The `__array_priority__` is there to handle cases when we do algebra between a `DataContainer` and a `BlockDataContainer`. In this case `BlockDataContainer` knows what to do while the `DataContainer` doesn't.

In this case I suggest to use `sapyb`.
```suggestion
            return self.sapyb(1,other,-1, out=kwargs.get('out', None})
```